### DISCUSSION TITLE
only show files of specific filetype

### DISCUSSION THREAD

**Discussion by exformation** [nvim-neo-tree/neo-tree.nvim#1624] [Q&A]
I'm trying to write a toggle that will filter out all files that **aren't** of the current filetype. 

I've messed around with various forms of the following, which is attempting to hide anything that isn't a lua file, but can't get it working: 
```
filesystem = {
  filtered_items = {
    hide_by_pattern = { '!(*.lua)' },
  }
}
```
I've also tried this to no avail: 
```
never_show_by_pattern = { '*' },
always_show_by_pattern = { '*.lua' },
```
I'm also not sure how I'd implement this as a toggle once I get it working.

**Comment by MeneerJansen** [comment]
Looks like we can't use regex in lua. See: https://stackoverflow.com/questions/6192137/how-to-write-this-regular-expression-in-lua

Maybe the developer of NeoTree can tell us if we can use the expression "not" which is sometimes expressed as ! (exclamation mark). Or may be there's a command like `only_show_by_pattern`? 

I don't speak lua so I don't know pattern matching works in lua.


P.S. Thank you for your post. I didn't know about `hide_by_pattern` and I used your code abouve to hide backup files in Linux (which end in ~) like so:
```
-- Let NeoTree hide ~ files (backup files in Linux).
require('neo-tree').setup {
  filesystem = {
    filtered_items = {
      hide_by_pattern = {"*~"},       
    },
},}
```
